src/librustdoc/html/render.rs

```diff
@@ -2268,9 +2286,12 @@ fn item_struct(w: &mut fmt::Formatter, cx: &Context, it: &clean::Item,
         if fields.peek().is_some() {
             write!(w, "<h2 class='fields'>Fields</h2>")?;
             for (field, ty) in fields {
-                write!(w, "<span id='{shortty}.{name}' class='{shortty}'><code>{name}: {ty}</code>
-                           </span><span class='stab {stab}'></span>",
-                       shortty = ItemType::StructField,
+                write!(w, "<span id='{item_type}.{name}' class='{item_type}'>
+                           <span id='{name}.{name_space}' class='invisible'>
```

Comment:
You'll need to use `derive_id` for these as well. The existing ids are also missing a call to `derive_id` so you may as well also fix that while you're at it.